src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -39,7 +39,7 @@ pub struct DynamicHoneyBadger<C, N: Rand> {
     pub(super) incoming_queue: Vec<(N, Message<N>)>,
     /// A random number generator used for secret key generation.
     // Boxed to avoid overloading the algorithm's type with more generics.
-    pub(super) rng: Box<dyn rand::Rng + Send + Sync>,
+    pub(crate) rng: Box<dyn rand::Rng + Send + Sync>,
```

Comment:
I'd prefer having the user pass another RNG (or the whole `TransactionQueue`?) into `QueueingHoneyBadgerBuilder` instead: that makes it more transparent which one is used for the queue and which one e.g. for DKG. (Although eventually we'll move to not storing them as fields at all anymore.)

Also, ideally each algorithm's code should serve as an example for a user who wants to build something similar: e.g. a user who wants to use a different kind of queue should be able to look at the QHB code to build their own implementation based on DHB, so it shouldn't use anything from DHB that's only available within the crate.